lib/Basics/ReadWriteSpinLock.cpp

```diff
@@ -100,28 +96,14 @@ void ReadWriteSpinLock::lockWrite() noexcept {
 }
 
 bool ReadWriteSpinLock::lockWrite(std::size_t maxAttempts) noexcept {
-  if (tryLockWrite()) {
-    return true;
-  }
-
-  uint64_t attempts = 0;
-
   // the lock is either hold by another writer or we have active readers
   // -> announce that we want to write
-  auto state = _state.fetch_add(::QueuedWriterIncrement, std::memory_order_relaxed);
-  while (++attempts < maxAttempts) {
-    while ((state & ~::QueuedWriterMask) == 0) {
-      // try to acquire lock and perform queued writer decrement in one step
-      if (_state.compare_exchange_weak(state, (state - ::QueuedWriterIncrement) | ::WriteLock,
-                                       std::memory_order_acquire)) {
-        return true;
-      }
-      if (++attempts > maxAttempts) {
-        return false;
-      }
+  uint64_t attempts = 0;
+  while (++attempts <= maxAttempts) {
+    if (tryLockWrite()) {
```

Comment:
This will no longer prefers writers over readers

Reply:
Fixed